reader_concurrency_semaphore.cc

```diff
@@ -1500,6 +1507,12 @@ void reader_concurrency_semaphore::maybe_admit_waiters() noexcept {
         auto& permit = _wait_list.front();
         dequeue_permit(permit);
         try {
+            // Abort the read if the remaining time the read has is less than half of its initial timeout
+            if (permit.timeout() - db::timeout_clock::now() <= (permit.timeout() - permit.created())/2) {
```

Comment:
This is not the initial timeout, this is just the timeout at the point the read arrived to the semaphore. This is fine, but I think the commend and patchlog should be adjusted.